smarts/sstudio/genhistories.py

```diff
@@ -54,14 +62,82 @@ def __init__(self, dataset_spec: Dict[str, Any], output: str):
         self._log = logging.getLogger(self.__class__.__name__)
         self.check_dataset_spec(dataset_spec)
         self._output = output
-        self._path = dataset_spec["input_path"]
+        self._path = os.path.expanduser(dataset_spec["input_path"])
         real_lane_width_m = dataset_spec.get("real_lane_width_m", DEFAULT_LANE_WIDTH)
-        lane_width = dataset_spec.get("map_net", {}).get(
-            "lane_width", real_lane_width_m
-        )
+        lane_width = self._map_spec.get("lane_width", real_lane_width_m)
         self._scale = lane_width / real_lane_width_m
         self._flip_y = dataset_spec.get("flip_y", False)
         self._swap_xy = dataset_spec.get("swap_xy", False)
+        # most trajectory datasets have .1s time delta (i.e., were collected at 10 Hz)
+        self._dt_sec = 0.1
+
+    class _WindowedReader:
+        """Iterates over the rows in file using a sliding window that keeps track of
+        both a number of rows before the current row and a number of rows after.
```

Comment:
Maybe this should be moved to `smarts.core.utils`.

Reply:
Yeah, I considered that.  I couldn't think of anywhere else we might use it in the future though.  But maybe I should just do it anyway.